### DISCUSSION TITLE
Graph time range and reports on the main screen

### DISCUSSION THREAD

**Discussion by mmip-labs** [akvorado/akvorado#1220] [Q&A]
Hi everybody!

I have couple questions about main screen (home page) in akvorado. 

1. For some reason a can't see reports about Autonomous systems and Countries, although, i can see this information in messages. Why?
2. Is it possible to change time range in graph on the main screen? I found only these setting for main screen (home page):

```
   homepage-top-widgets:
        - src-as
        - src-port
        - protocol
        - src-country
        - etype
      homepagegraphfilter: InIfBoundary = 'external'

```

![2024-05-09_14-29-31](https://github.com/akvorado/akvorado/assets/63446512/91df8728-fce7-4f14-a6d2-74c3bc149a63)

**Comment by vincentbernat** [comment]
Top source countries have a filter applied of `InIfBoundary = external`. Same for top source AS. It should be the same for source ports, this is something missing (fixed in 64acf78af0be).

Setting up the time range for the graph is not possible yet. See 1037d66d223f for next version.

**Comment by mmip-labs** [comment]
I see. In my messages **InIfBoundary = internal**. How can i change this behavior to external? In my case, i have two NetFlow exporters on the edge of network (ingress and egress traffic).

**Reply by vincentbernat** [reply]
This is set from interface classification. You need to configure it correctly for most flows to either be internal or external.

**Reply by mmip-labs** [reply]
Here?

core:
  workers: 6
  default-sampling-rate: 1000
  exporter-classifiers:
    # This is an example. This should be customized depending on how
    # your exporters are named.
    - ClassifySiteRegex(Exporter.Name, "^([^-]+)-", "$1")
    - ClassifyRegion("moscow")
    - ClassifyTenant("mmip")
    - ClassifyRole("edge")
  interface-classifiers:
    # This is an example. This must be customized depending on the
    # descriptions of your interfaces. In the following, we assume
    # external interfaces are named "Transit: Cogent" Or "IX:
    # FranceIX".
    - |
      ClassifyConnectivityRegex(Interface.Description, "^(?i)(transit|pni|ppni|ix):? ", "$1") &&
      ClassifyProviderRegex(Interface.Description, "^\\S+?\\s(\\S+)", "$1") &&
      ClassifyExternal()
    - ClassifyInternal()

**Reply by mmip-labs** [reply]
I just left:

  interface-classifiers:
    # This is an example. This must be customized depending on the
    # descriptions of your interfaces. In the following, we assume
    # external interfaces are named "Transit: Cogent" Or "IX:
    # FranceIX".
    - |
      ClassifyExternal()

it works now, but i still don't understand logic, sorry (( Could you describe mo detailed, what **interface-classifiers** and **exporter-classifiers** do and how do they work. I have read https://demo.akvorado.net/docs/configuration#core but didn't understand. I think i'm just stupid (( Many thanks.


![2024-05-09_17-20-57](https://github.com/akvorado/akvorado/assets/63446512/5f6218a7-a06a-4b78-a39b-05e13b8ced2e)

**Comment by vincentbernat** [comment]
External interfaces are the one towards internet, internal interfaces are the ones towards your internal network. You need to classify all your interfaces into one of these categories using their names or descriptions.

**Comment by nneo16** [comment]
I've spent some time understand the logic of this section, and here what I found:

- Each line (start with "-") is a rule
- The "|" in the start of the line will joins the lines below it into one line (means one rule) 
- The "Regex" classifier is optional in case you want filter some strings into a group
- For exporters only: each rule can apply to all exporters (if it matches)
- For interfaces only: if one interface matches a rule, it can't applied to other rules (means one interface can only be either "internal" or "external")

Hope this can help :)
